File: Android/Sources/UserControlLibraries/GoogleMapsLib/src/main/java/com/genexus/android/controls/maps/google/MapLocationBounds.java
package com.genexus.android.controls.maps.google;

import com.genexus.android.core.controls.maps.common.IMapLocationBounds;
import com.google.android.gms.maps.model.LatLngBounds;

class MapLocationBounds implements IMapLocationBounds<MapLocation>
{
	private final LatLngBounds mBounds;

	public MapLocationBounds(LatLngBounds bounds)
	{
		mBounds = bounds;
	}

	LatLngBounds getLatLngBounds()
	{
		return mBounds;
	}

	@Override
	public MapLocation southwest()
	{
		return new MapLocation(mBounds.southwest);
	}

	@Override
	public MapLocation northeast()
	{
		return new MapLocation(mBounds.northeast);
	}
}

File: Android/Sources/UserControlLibraries/HMSMapKitLib/src/main/java/com/genexus/android/controls/maps/hms/MapLocationBounds.kt
package com.genexus.android.controls.maps.hms

import com.genexus.android.core.controls.maps.common.IMapLocationBounds
import com.huawei.hms.maps.model.LatLngBounds

internal class MapLocationBounds(val latLngBounds: LatLngBounds) : IMapLocationBounds<MapLocation> {
    override fun southwest(): MapLocation {
        return MapLocation(latLngBounds.southwest)
    }

    override fun northeast(): MapLocation {
        return MapLocation(latLngBounds.northeast)
    }
}
